docs/reference.cue

```diff
@@ -1 +1,36 @@
 package metadata
+
+#Fields: [Name=string]: #Fields | _
+
+remap: {
+  errors: [Name=string]: {
+    description: string
+    name: Name
+  }
+
+  functions: [Name=string]: {
+    arguments: [
+      ...{
+        required: bool,
+
+        if !required {
+          name: string
+        }
+
+        type: "float" | "int" | "string"
+      }
+    ]
+    category: "coerce" | "parse"
+    description: string
+    examples: [
+      {
+        title: string
+        input: #Fields
+        source: string
+        output: #Fields
+      },
+      ...
+    ]
+    name: Name
+  }
```

Comment:
We'll also want to define the return value of a function.

Once we have that, and the other points I raised are solved, we could render documentation that makes it easy to quickly see the signature of a function, such as:

```rust
to_int(value = (int | float | string | bool | timestamp), [default = int]) -> int
```